### PR TITLE
Converted submission transaction to task (EXPOSUREAPP-2482)

### PR DESCRIPTION
**Pull request by chris-cwa**

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionTask.kt [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    ) : TaskFactory<DefaultProgress, Task.Result> {
+
+        override val config: TaskFactory.Config = Config()
+        override val taskProvider: () -> Task<DefaultProgress, Task.Result> = {
+            taskByDagger.get()
+        }
+    }
+
+    companion object {
+        private val SUBMISSION_TASK_TIMEOUT = Duration.standardMinutes(8)
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionTask.kt:110]: Can we inline this?

**Review by d4rken — COMMENTED** [review]
Few readability improvements.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionTask.kt [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                    registrationToken,
+                    getHistory(),
+                    true,
+                    supportedCountries()
+                ).also {
+                    checkCancel()
+                }
+            )
+        }
+        SubmissionService.submissionSuccessful()
```
Comment (kolyaopahle) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionTask.kt:46]: As a note for further refactoring, i really don't like the `SubmissionService` in its current state. Its just a agglomeration of grown over time functions with no real central goal. Id really like to spend some time here but obviously not in this PR.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionTask.kt (hunk 1 above)
Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionTask.kt:110]: I agree, its only used in one place and the variable that its assigned to is already descriptive enough in my opinion

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionTask.kt [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                toNewConfig {
+                    addSupportedCountries(FALLBACK_COUNTRY)
+                }.also {
+                    Timber.w("Country list was empty, corrected: %s", it.supportedCountriesList)
+                }
+            } else {
+                this
+            }.supportedCountriesList
+        }
+    }
```
[the remark this answers is not in the document]
Comment (chris-cwa) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionTask.kt:75]: simplified this part

**Review by SamuraiKek — DISMISSED** [review]
Lgtm. Code is readable and easy to follow.

**Comment by BMItter** [comment]
I just tested the submission flow. Negativ/Positive tests were all right, but after putting in symtoms (Keine Angaben) I got a missingTaskFactoryException (No Task factory mapped for class). I will send you the report internally.

**Review by BMItr — APPROVED** [review]
lgtm :muscle:, above mentioned exception is gone, passed submission tests.

**Review by SamuraiKek — APPROVED** [review]

**Pull request merged**